spacy/strings.pyx

```diff
@@ -110,28 +129,35 @@ cdef class StringStore:
         string_or_id (bytes, str or uint64): The value to encode.
         Returns (str / uint64): The value to be retrieved.
         """
-        if isinstance(string_or_id, str) and len(string_or_id) == 0:
-            return 0
-        elif string_or_id == 0:
-            return ""
-        elif string_or_id in SYMBOLS_BY_STR:
-            return SYMBOLS_BY_STR[string_or_id]
-        cdef hash_t key
+        cdef hash_t hash
+        cdef Utf8Str* utf8str = NULL
+
         if isinstance(string_or_id, str):
-            key = hash_string(string_or_id)
-            return key
+            if len(string_or_id) == 0:
+                return 0
+
+            symbol = SYMBOLS_BY_STR.get(string_or_id, None)
```

Comment:
The default is `None` here so that the `hash_string` call is only made if required, right? I would leave a comment about that.

Reply:
Correct; will do.